### ISSUE TITLE
Project/Component name and description

### ISSUE THREAD

**Issue by massimobio** [celanajaya/AdjustableTextField#1]
Not critical but we should brainstorm a better name for the project/component, which will be important for discoverability if we add support for Cocoapods and when we make a nice README about it.

The current name has some issues:
- AdjustableTextField puts a lot of importance on the TextField aspect which implies any kind of text can be inserted while it's really for numeric values only
- The 'adjustable' adjective in this combination seems to indicate that you can adjust the textfield somehow... size? color?... unclear.

If I try to describe the component in words, it's a numeric control that allows both typing-in numeric values as well as click-n-draging to adjust the value.
Garageband has a nice implementation of this kind of control:
<img width="329" alt="screen shot 2017-07-26 at 10 46 48 am" src="https://user-images.githubusercontent.com/240599/28627460-dfb0511a-71ef-11e7-9706-59068a8dde0f.png">

The tempo value looks like a label but when you hover over it, the background highlights so it feels clickable. When you click on it, the cursor turns into a short dash which helps with discoverability of the drag-to-adjust feature. If you double click it turns into an editable textfield with the numbers selected so that you can immediately start typing.

Name brainstorming:
- it's a numeric control/label/button
- it's a type-your-value in a textfield control
- it's a (hidden) slider kind of control
- don't think SwiftySomething is good because it's not a port of an obj-c project and there are too many Swifty projects already
...
NumericMultiControl 
AdvancedNumericControl
Value something could also be good
Number instead of Numeric....
...
...

**Comment by celanajaya** [comment]
Yes, totally. Should we also mention the click-and-drag part?

**Comment by massimobio** [comment]
yeah, the click-and-drag part is what makes me think of a slider, cause that's what you do when you want to adjust a slider.

> There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors.

ClickAndDragNumericControl 
NumericDualControl
ClickAndDragAndTypeNumericControl
DualInputNumberControl
MultiImputNumberField
ClickAndDragNumericField
ValueMultiControl

these inspiring some other one for you or see one that sticks out to you?

**Comment by massimobio** [comment]
While writing this [comment at point 2.](https://github.com/celanajaya/AdjustableTextField/issues/6#issuecomment-318837104), my thinking evolved towards a less specific name.

Think of it as something in line with one of these:
NumberInput
NumericField
NumberValueControl
ValueInput
...

that offers maxValue minValue and currently two input methods: click-n-drag and type your value (with validation)

In the future we could to add more input options like a + - stepper for example as I have here:
<img width="148" alt="screen shot 2017-07-29 at 11 18 12 am" src="https://user-images.githubusercontent.com/240599/28745881-be59c5b6-744f-11e7-9e21-86e2b3fa1ebd.png">

**Comment by celanajaya** [comment]
How about NumericControlField?
Also I think adding a stepper would be very cool.

**Comment by massimobio** [comment]
I like it!
Renaming a project is 😱 but it might be ok at this scale.
If you've never done it I'll leave the fun for you 😉

Once all the PR's are merged, you'd want to rename the project, the target, all the necessary file names, all the occurrences of AdjustableTextFiled (class, delegate...) and the project name here in GitHub. you can just do it all in Master, simpler than attempting a PR for something so encompassing.

<img width="1478" alt="screen shot 2017-08-03 at 9 41 43 am" src="https://user-images.githubusercontent.com/240599/28924633-6b85fc8a-7830-11e7-917e-67bc611ccf33.png">
